### PR TITLE
feat: support for Postgres15

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Support for Postgres 15 and added tests.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: tests/other_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func TestRunRestoredDBExample(t *testing.T) {
 		BestRegionYAMLPath: regionSelectionPath,
 		ResourceGroup:      resourceGroup,
 		TerraformVars: map[string]interface{}{
-			"pg_version": "13",
+			"pg_version": "14",
```
Comment (ocofaigh) [inline tests/other_test.go:21]: The PITR test is testing version 14, so set this back to 15 and we should then have 1 test for every version

File: variables.tf [hunk 2]
```diff
@@ -31,12 +31,13 @@ variable "pg_version" {
...
     condition = anytrue([
       var.pg_version == null,
+      var.pg_version == "15",
       var.pg_version == "14",
       var.pg_version == "13",
       var.pg_version == "12",
       var.pg_version == "11"
     ])
-    error_message = "Version must be 11 or 12 or 13 or 14. If no value passed, the current ICD preferred version is used."
+    error_message = "Version must be 11 or 12 or 13 or 14 or 15. If no value passed, the current ICD preferred version is used."
```
Comment (ocofaigh) [inline variables.tf:40]: This reads better: `Version must be 11, 12, 13, 14 or 15`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see minor comments

File: tests/other_test.go (hunk 1 above)
Comment (rajatagarwal-ibm) [reply tests/other_test.go:21]: Changed that to 13 again, as:

FSCloud: 15
PITR: 14
RestoreDB: 13
Basic example: 12
Upgrade test: 11

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**